app/models/friend_request.rb

```diff
@@ -1,7 +1,30 @@
+# frozen_string_literal: true
+
 class FriendRequest < ApplicationRecord
+  enum status: { pending: 'pending', accepted: 'accepted', declined: 'declined',
+                 cancelled: 'cancelled' }
+
   validates :sender, presence: true
   validates :receiver, presence: true
+  validates :status, presence: true
```

Comment:
Similar to what I did in #4 I created `ryto` and `mike`. In this case, no bug is happening but I just feel it's gonna be simpler to make a friend request without having to validate the `status` and just make the `status` to have a default value of `false`. As it sounds weird to have a `status` of false, I would then change the name of this attribute to `confirmed`. 

So then whenever I create a friend request, until the time it's confirmed it will have a value of `false` in the `friend_requests` table in the `db`. 

While as I mentioned, I won't touch the controllers yet now. I just wanna give you this tip. Later on in the `friendships_controller` and `friend_requests_controller` you will have simpler actions if we put more of the logic of changing this `confirmed` status here in the model. In those controllers later on you might just want to call a method written in your `friend_request` model that changes the `confirmed` status into `true` once a `friendship` record is created (`friendships_controller` `create` action).

Ok, I am getting ahead of myself here. But the point of this particular comment is to just have a default `confirmed` of false to a `friend_request` record then just update that later on (as a method in the model).

Reply:
The main purpose of using the friend request statuses(accepted, declined, canceled and pending). I think in some future, there would be some reasons to block users which is declined by many times. Such as Burhan sends 3 times friend requests to Ryan and Ryan declines all of these requests in a sequence. So, we can block the declined user in an easier way. I will put these handling mechanisms through FriendRequestHandler service.

Reply:
It's a native database type, check [migration file](https://github.com/btuerker/odin-facebook/blob/controller/db/migrate/20190520101528_change_friend_request_status_to_typed_value.rb)